### PR TITLE
teal-language-server: dev-1 -> 0.0.5.1

### PR DESCRIPTION
**Pull request by barsoosayque**
I have regenerated `teal-language-server`, `tl` and a new `lusc_luv` luarocks packages, since previous teal language server didn't work for me. I had to also update `tl`, which introduced a lock file that just failed to build the package. Also updated `cyan`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/lua-modules/generated-packages.nix [hunk 1]
```diff
@@ -2443,6 +2443,31 @@ buildLuarocksPackage {
...
+  pname = "lusc_luv";
+  version = "4.0.1-1";
+  knownRockspec = (fetchurl {
+    url    = "mirror://luarocks/lusc_luv-4.0.1-1.rockspec";
+    sha256 = "1bgk481ljfy8q7r3w9z1x5ix0dm6v444c7mf9nahlpyrz9skxakp";
+  }).outPath;
+  src = fetchFromGitHub {
+    owner = "svermeulen";
+    repo = "lusc_luv";
+    rev = "main";
```
Comment (steeleduncan) [inline pkgs/development/lua-modules/generated-packages.nix:2457]: The `main` branch will eventually point to a different commit, the hash would be wrong and the build will break. For reproducibility you should use a tag

File: pkgs/development/lua-modules/generated-packages.nix [hunk 2]
```diff
@@ -3280,22 +3305,22 @@ buildLuarocksPackage {
...
-    sha256 = "01l44c6bknz7ff9xqgich31hlb0yk4ms5k1hs4rhm3cs95s5vlzc";
+    url    = "mirror://luarocks/teal-language-server-0.0.5-1.rockspec";
+    sha256 = "11ps1hgkgli4sf9gcj7pin4kbc5w0yck0daig1ghqssn2q9m2x5l";
   }).outPath;
   src = fetchFromGitHub {
     owner = "teal-language";
     repo = "teal-language-server";
-    rev = "67b5d7cad60b9df472851a2c61591f2aab97da47";
-    hash = "sha256-fUuOjJrwpLU1YoJm3yn+X15ioRf4GZoi6323On1W2Io=";
+    rev = "main";
```
Comment (steeleduncan) [inline pkgs/development/lua-modules/generated-packages.nix:3319]: same here, it needs pinning

File: pkgs/development/lua-modules/generated-packages.nix [hunk 3]
```diff
@@ -3378,22 +3403,22 @@ buildLuarocksPackage {
...
-    sha256 = "15p67r5bjp997pymjq80yn1gyf7r5g2nwkachkwx88100ihblqrc";
+    url    = "mirror://luarocks/tl-0.24.1-1.rockspec";
+    sha256 = "0r9cp5w824mmn3hmcwjcga182sa25hyvnkk81025hn49jfgr2hps";
   }).outPath;
   src = fetchFromGitHub {
     owner = "teal-language";
     repo = "tl";
-    rev = "v0.15.3";
-    hash = "sha256-nkwPYI4uB1rTtcBsZ7TKNPusWXtXViyBDSkiL9UH+Wo=";
+    rev = "v0.24.1";
```
Comment (steeleduncan) [inline pkgs/development/lua-modules/generated-packages.nix:3417]: `refs/tags/v0.24.1`

**Review by steeleduncan — CHANGES_REQUESTED** [review]

**Comment by steeleduncan** [comment]
Additionally, the commits should be squashed to follow the normal pattern

File: pkgs/development/lua-modules/overrides.nix [hunk 4]
```diff
@@ -855,6 +855,12 @@ in
     ];
   });
 
+  tl = prev.tl.overrideAttrs (_: {
```
Comment (teto) [inline pkgs/development/lua-modules/overrides.nix:858]: you can now do `overrideAttrs({` without the `_:`

**Comment by teto** [comment]
@barsoosayque thanks for the bump. I am curious how you use teal and why ? notably luarocks has been rewritten in teal recently, would you be interested in trying to bump the luarocks package ?

**Comment by barsoosayque** [comment]
> @barsoosayque thanks for the bump. I am curious how you use teal and why ? notably luarocks has been rewritten in teal recently, would you be interested in trying to bump the luarocks package (later in another PR)?

Teal is a great tool for game scripting, as well as just plain lua, but I enjoy the features of teal. So I'm only interested in `teal-language-server` because it provides language server for our game scripts, I'm not actually running/building tl packages.

As for the changes, I'll try to make fixes on Saturday. I can also probably take a look at bumping `luarocks`, though the script for updating was broken for me the last time I checked, so it's a pain to update :pensive:

**Comment by teto** [comment]
thanks for the answer. My experience with teal was not great and I've seen people trying to use it and give up. I am always curious how people use the ecosystem, it helps knowing if changes in the nix packages are going to affect users.

> I can also probably take a look at bumping luarocks, though the script for updating was broken for me the last time I checked, so it's a pain to update 😔

That would be about luarocks_bootstrap pkgs/development/tools/misc/luarocks/default.nix, which bypasses the plugins updater. I am not excited about dealing with all the teal changes so finding someone versed in teal and interested in contributing to nixpkgs would be the dream. But dont bother if you are not interested ;)

**Comment by teto** [comment]
I was going to squash and merge but it seems to break other packages. Could you fix those please ? it probably means updating them as well
```
6 packages failed to build:
lua51Packages.cyan luaPackages.cyan lua53Packages.cyan lua54Packages.cyan luajitPackages.cyan vimPlugins.nvim-teal-maker

15 packages built:
lua51Packages.lusc_luv lua51Packages.teal-language-server lua51Packages.tl luaPackages.lusc_luv luaPackages.teal-language-server luaPackages.tl lua53Packages.lusc_luv lua53Packages.teal-language-server lua53Packages.tl lua54Packages.lusc_luv lua54Packages.teal-language-server lua54Packages.tl luajitPackages.lusc_luv luajitPackages.teal-language-server luajitPackages.tl
```

**Comment by barsoosayque** [comment]
I just force-pushed new commits, tried my best to follow the contributor guide. Updated `cyan`, and it fixed  `vimPlugins.nvim-teal-maker`. I also ran `nixpkgs-review` and there seems to be no more build errors.

**Comment by teto** [comment]
passed nixpkgs-review for me as well

**Pull request merged**